plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -263,9 +270,20 @@ static ConnectorPageSource createPageSource(
             MessageColumnIO messageColumn = getColumnIO(fileSchema, requestedSchema);
 
             Map<List<String>, ColumnDescriptor> descriptorsByPath = getDescriptors(fileSchema, requestedSchema);
-            TupleDomain<ColumnDescriptor> parquetTupleDomain = options.isIgnoreStatistics()
+
+            // Combine static and dynamic predicates
+            TupleDomain<HiveColumnHandle> staticPredicate = hudiSplit.getPredicate();
+            TupleDomain<HiveColumnHandle> dynamicPredicate = dynamicFilter.getCurrentPredicate()
+                    .transformKeys(HiveColumnHandle.class::cast);
+            TupleDomain<HiveColumnHandle> combinedPredicate = staticPredicate.intersect(dynamicPredicate);
+
+            if (!combinedPredicate.isAll()) {
+                log.debug("Combined predicate for Parquet read (Split: %s): %s", hudiSplit, combinedPredicate);
+            }
```

Comment:
nit: wrap this logic into a method so it's only called for predicate pushdown with `getParquetTupleDomain` and easier to understand?